packages/lesmis-server/src/routes/TranslatableRoute.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+
+// Overwrite the respond function to include a translate step
+// Use translationKeys to define which pieces of the response body
+// are translated for a route
+
+type StringKey = {
+  type: 'string';
+}
+
+type ObjectKey = {
+  type: 'object';
+  properties: { [key: string]: TranslationKey };
+}
+
+type ArrayKey = {
+  type: 'array';
+  items: TranslationKey;
+  where?: (entry: any) => boolean;
+}
+
+type TranslationKey = StringKey | ObjectKey | ArrayKey;
+type TranslationValue = undefined | string | TranslationValue[] | { [key: string]: TranslationValue };
+
+export class TranslatableRoute extends Route {
+  translationSubGroup: string = '';
+  translationKeys: TranslationKey = { type: 'string' };
+
+  respond(responseBody: TranslationValue, statusCode: number) {
+    const translatedResponse = this.translateResponse(this.translationKeys, responseBody);
+    super.respond(translatedResponse, statusCode);
+  }
+
+  // Overload this function so that translationKey implies the shape of translationValue
+  translateResponse(translationKey: StringKey, translationValue: string): TranslationValue;
+  translateResponse(translationKey: ObjectKey, translationValue: { [key: string]: TranslationValue }): TranslationValue;
+  translateResponse(translationKey: ArrayKey, translationValue: TranslationValue[]): TranslationValue;
+  translateResponse(translationKey: TranslationKey, translationValue: TranslationValue): TranslationValue;
+  translateResponse(translationKey: any, translationValue: any): any {
+    if (!translationValue) {
+      return undefined;
+    }
+    switch(translationKey.type) {
+      case 'string':
+        return this.res.__(`${this.translationSubGroup}.${translationValue}:${translationValue}`);
```

Comment:
That's probably a handy comment tbh, it would be hard to pick up without going to the docs.

An unrelated point: I'm not a fan of the `__` mystery function

Can we please use the `api` config option? From the docs:
```
// hash to specify different aliases for i18n's internal methods to apply on the request/response objects (method -> alias).
  // note that this will *not* overwrite existing properties with the same name
  api: {
    __: 't', // now req.__ becomes req.t
    __n: 'tn' // and req.__n can be called as req.tn
  },
```